be/src/storage/lake/transactions.cpp

```diff
@@ -153,28 +202,35 @@ StatusOr<TabletMetadataPtr> publish_version(TabletManager* tablet_mgr, int64_t t
     return new_metadata;
 }
 
-Status publish_log_version(TabletManager* tablet_mgr, int64_t tablet_id, int64_t txn_id, int64_t log_version) {
-    auto txn_log_path = tablet_mgr->txn_log_location(tablet_id, txn_id);
-    auto txn_vlog_path = tablet_mgr->txn_vlog_location(tablet_id, log_version);
-    // TODO: use rename() API if supported by the underlying filesystem.
-    auto st = fs::copy_file(txn_log_path, txn_vlog_path);
-    if (st.is_not_found()) {
-        ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(txn_vlog_path));
-        auto check_st = fs->path_exists(txn_vlog_path);
-        if (check_st.ok()) {
-            return Status::OK();
-        } else {
-            LOG_IF(WARNING, !check_st.is_not_found())
-                    << "Fail to check the existance of " << txn_vlog_path << ": " << check_st;
+Status publish_log_version(TabletManager* tablet_mgr, int64_t tablet_id, const int64_t* txn_ids,
+                           const int64_t* log_versions, int txns_size) {
+    std::vector<std::string> files_to_delete;
+    for (int i = 0; i < txns_size; i++) {
+        auto txn_id = txn_ids[i];
+        auto log_version = log_versions[i];
+        auto txn_log_path = tablet_mgr->txn_log_location(tablet_id, txn_id);
+        auto txn_vlog_path = tablet_mgr->txn_vlog_location(tablet_id, log_version);
+        // TODO: use rename() API if supported by the underlying filesystem.
+        auto st = fs::copy_file(txn_log_path, txn_vlog_path);
+        if (st.is_not_found()) {
+            ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(txn_vlog_path));
+            auto check_st = fs->path_exists(txn_vlog_path);
+            if (check_st.ok()) {
+                continue;
+            } else {
+                LOG_IF(WARNING, !check_st.is_not_found())
+                        << "Fail to check the existance of " << txn_vlog_path << ": " << check_st;
+                return st;
+            }
+        } else if (!st.ok()) {
             return st;
+        } else {
+            files_to_delete.emplace_back(txn_log_path);
+            tablet_mgr->metacache()->erase(txn_log_path);
         }
-    } else if (!st.ok()) {
-        return st;
-    } else {
-        delete_files_async({txn_log_path});
-        tablet_mgr->metacache()->erase(txn_log_path);
-        return Status::OK();
     }
+    delete_files_async(std::move(files_to_delete));
```

Comment:
Don't we need to deal with issues like `publish_version` has that going from batch publish mode back to single publish mode?

Reply:
The issues has been avoided by not removed the txn_logs when batch publish